src/vma/dev/ring_eth_cb.cpp

```diff
@@ -712,8 +712,14 @@ ring_eth_cb::~ring_eth_cb()
 
 	// explicitly destroy the qp and cq before this destructor finshes
 	// since it will release the memory allocated
-	delete m_p_qp_mgr;
-	m_p_qp_mgr = NULL;
+	if (m_p_qp_mgr) {
+		// 'down' the active QP/CQ
+		m_p_qp_mgr->down();
```

Comment:
this will probably not a good idea
this function will call trigger_completion_for_all_sent_packets() that is an invalid flow for this ring.
did you test this?